.github/actions/sil-kit-ci/run/action.yml

```diff
@@ -0,0 +1,114 @@
+name: sil-kit-ci/run
+
+description: Run the SIL Kit CI builds.
+
+inputs:
+
+  preset-name:
+    description: The preset name from CMakePresets.json
+    required: true
+
+  linux-docker-image:
+    description: Docker image used under Linux
+    required: false
+    default: ghcr.io/mariusbgm/sil-kit-ci-ubuntu-18.04:main
+
+  linux-c-compiler:
+    description: C compiler used under Linux
+    required: false
+    default: gcc-8
+
+  linux-cxx-compiler:
+    description: C++ compiler used under Linux
+    required: false
+    default: g++-8
+
+  msvc-arch:
+    description: Target architecture used under Windows/MSVC
+    required: false
+    default: 'x64'
+
+  msvc-toolset:
+    description: Target toolset used under Windows/MSVC
+    required: false
+    default: '14.1'
+
+  upload-artifacts:
+    description: Determines if artifacts created by the actions should be uploaded
+    required: false
+    default: 'false'
+
+  retention-days:
+    description: Determines the number of days any uploaded artifacts should be retained
+    required: false
+    default: '1'
+
+  package-name:
+    description: Name of the package to upload as an artifact
+    required: false
+    default: 'sil-kit.zip'
+
+runs:
+  using: composite
+  steps:
+
+    - name: setup
+      uses: ./.github/actions/sil-kit-ci/setup
+      id: setup
+      with:
+        preset-name: ${{ inputs.preset-name }}
+        linux-docker-image: ${{ inputs.linux-docker-image }}
+        msvc-arch: ${{ inputs.msvc-arch }}
+        msvc-toolset: ${{ inputs.msvc-toolset }}
+
+    - name: (linux) select c compiler
+      id: linux-cmake-configure-args-c-compiler
+      if: runner.os == 'Linux'
+      shell: bash
+      run: echo 'cmake-configure-args=-DCMAKE_C_COMPILER=${{ inputs.linux-c-compiler }}' >> "$GITHUB_OUTPUT"
+
+    - name: (linux) select c++ compiler
+      id: linux-cmake-configure-args-cxx-compiler
+      if: runner.os == 'Linux'
+      shell: bash
+      run: echo 'cmake-configure-args=-DCMAKE_CXX_COMPILER=${{ inputs.linux-cxx-compiler }}' >> "$GITHUB_OUTPUT"
+
+    - name: configure cmake project
+      uses: ./.github/actions/sil-kit-ci/execute
+      with:
+        preset-name: ${{ inputs.preset-name }}
+        linux-docker-image: ${{ inputs.linux-docker-image }}
+        command: cmake
+        args: --preset "${{ inputs.preset-name }}" ${{ join(steps.*.outputs.cmake-configure-args, ' ') }}
```

Comment:
The steps that produce the `cmake-configure-args` could be put into a combined action, that produces a single output `cmake-configure-args` with the exact same expression (similar to the `sil-kit-ci/dirs` action). Then the runner-specific steps can be 'hidden' behind a single action call.